consensus/reactor.go

```diff
@@ -107,14 +107,17 @@ func (conR *Reactor) OnStop() {
 func (conR *Reactor) SwitchToConsensus(state sm.State, skipWAL bool) {
 	conR.Logger.Info("SwitchToConsensus")
 
-	// We have no votes, so reconstruct LastCommit from SeenCommit.
+	// We need to lock, as we are not entering consensus state from State's `handleMsg` or `handleTimeout`
+	conR.conS.mtx.Lock()
+	// We have no votes, so reconstruct LastCommit from SeenCommit
 	if state.LastBlockHeight > 0 {
 		conR.conS.reconstructLastCommit(state)
 	}
 
 	// NOTE: The line below causes broadcastNewRoundStepRoutine() to broadcast a
 	// NewRoundStepMessage.
 	conR.conS.updateToState(state)
+	conR.conS.mtx.Unlock()
```

Comment:
reconstructLastCommit and updateToState can panic, which will leave conS.mtx in a locked state, producing deadlock for any concurrent lockers.

Reply:
@08d2, are you deeply familiar with the `SwitchToConsensus` function and where it's called? If so, please point out where, exactly (which files/lines of code), you see `conS.mtx` being locked concurrently during this specific `SwitchToConsensus` call.

Reply:
By design, CometBFT never tries to recover from a panic (crash failures always being preferable to general byzantine behaviour in both crash-tolerant and BFT algorithms), so, strictly speaking, this is a non-issue.
Nevertheless, I've changed the locking I had added to a RAII-style

Reply:
> @08d2, are you deeply familiar with the `SwitchToConsensus` function and where it's called? If so, please point out where, exactly (which files/lines of code), you see `conS.mtx` being locked concurrently during this specific `SwitchToConsensus` call.

My mistake for qualifying the error as only for "concurrent callers". This was not accurate. If any caller, concurrent or otherwise, were to panic in the ways I described previously, the conS mutex would deadlock for all subsequent callers.